### ISSUE TITLE
Bootstrap failed when creating instances

### ISSUE THREAD

**Issue by msangoi** [eclipse/wakaama#241]
Since commit 35bc79c, some object creation requests failed with 4.00 bad request: 

```
5684      CoAP     99     CON, MID:9554, POST, TKN:52 25 ff 25, /bs?ep=TEST
56830     CoAP     80     CON, MID:46747, DELETE, TKN:21 29 04 9f ab
5684      CoAP     80     ACK, MID:46747, 2.02 Deleted, TKN:21 29 04 9f ab
56830     CoAP     79     ACK, MID:9554, 2.04 Changed, TKN:52 25 ff 25
56830     CoAP     191    CON, MID:46748, PUT, TKN:ce 5e 60 52, /0/0
5684      CoAP     79     ACK, MID:46748, 2.04 Changed, TKN:ce 5e 60 52
56830     CoAP     233    CON, MID:46749, PUT, TKN:60 04 31 6a ad, /0/1
5684      CoAP     80     ACK, MID:46749, 4.00 Bad Request, TKN:60 04 31 6a ad
56830     CoAP     103    CON, MID:46750, PUT, TKN:b6 6b 94, /1/0
5684      CoAP     78     ACK, MID:46750, 4.00 Bad Request, TKN:b6 6b 94
```

It seems to be related to this change: https://github.com/eclipse/wakaama/commit/35bc79c80312028b1eef837d56e9c68dc1db7f1b#diff-bdcb12ba8eb5875cbb4a8c4085f29d77R297

I guess this change was made for the CREATE requests of the device management interface. But it is not valid for the bootstrap WRITE requests.

And, from the above exchanges, I don't know why the first WRITE request for path /0/0 is successful and the next ones are not.

**Comment by dnav** [comment]
Indeed.
For the management interface the check is already done in dm_handleRequest().

See #242 for a proposed fix.

Regards,

**Comment by sbernard31** [comment]
Your fix seems ok for the Bad request issue.
But how do you explain that the first bootstrap write request succeed ? I suspect this reveal another bug ? uri-> flash seems to set correctly for the path /0/0. WDYT ?

**Comment by dnav** [comment]
I could not reproduce this with my set-up.
I think that the instance 0 was already existing in the Client so object_write() was called instead of object_create().

Regards,

**Comment by sbernard31** [comment]
This make sense.

**Comment by msangoi** [comment]
> I think that the instance 0 was already existing in the Client so object_write() was called instead of object_create()

It should not be the case after the initial delete '/' :)

**Comment by dnav** [comment]
@msangoi I just checked this. /0/0 is the Bootstrap Server Account on the wakaama example client thus it is not affected by the Delete /.

Regards,

**Comment by msangoi** [comment]
Ok sorry, it looks good then. Thx for the quick fix!

**Comment by dnav** [comment]
Fixed by #242
